akka-actor/src/main/scala/akka/pattern/AskSupport.scala

```diff
@@ -328,7 +328,9 @@ private[akka] object PromiseActorRef {
   private case object Stopped
   private final case class StoppedWithPath(path: ActorPath)
 
-  def apply(provider: ActorRefProvider, timeout: Timeout, targetName: String, message: Any, sender: ActorRef = Actor.noSender): PromiseActorRef = {
+  private val ActorStopResult = Failure(new ActorKilledException("Stopped"))
+
+  def apply(provider: ActorRefProvider, timeout: Timeout, targetName: ⇒ String, message: Any, sender: ActorRef = Actor.noSender): PromiseActorRef = {
```

Comment:
Because most of the time the the target name is not used and deferring creation of that String until usage saves alot of stringbuilders and char[]